software_training_assignment/src/main.cpp

```diff
@@ -0,0 +1,107 @@
+/*
+The main client for the node 
+*/
+
+#include <ros/ros.h>
+#include <string>
+#include <math.h>
+
+//Service for turtlesim spawn 
+#include <turtlesim/Spawn.h> 
+
+//Service to clear the turtles 
+//(I am guessing it clears the turtles)
+//Based on the wording of the documentation
+#include <std_srvs/Empty.h> 
+
+//Including our custom message
+#include <software_training_assignment/turtleDistances.h> 
+
+//Include Action Library 
+#include <actionlib/client/simple_action_client.h> 
+
+//Class for determining the distances between the turtles
+class turtleDistance{
+    public:
+        int xd;
+        int yd;
+        int distance;
+    turtleDistance(int t1_x, int t1_y, int t2_x, int t2_y){
+       xd = t1_x - t2_x;
+       yd = t1_y - t2_y;
+       distance = sqrt(xd*xd + yd*yd); 
+    }
+}
+
+int main(int argc, char **argv) {
```

Comment:
this code only runs once. what happens if I want to send the turtle to different places more than once. you should have a setup portion to call clear and spawn and then a main loop to continuously listen and perform the action and update the turtle coordinates.